make-distribution.sh

```diff
@@ -188,6 +188,7 @@ echo "Build flags: $@" >> "$DISTDIR/RELEASE"
 # Copy jars
 cp "$SPARK_HOME"/assembly/target/scala*/*assembly*hadoop*.jar "$DISTDIR/lib/"
 cp "$SPARK_HOME"/examples/target/scala*/spark-examples*.jar "$DISTDIR/lib/"
+cp "$SPARK_HOME"/external/kafka/scala*/*kafka*assembly*.jar "$DISTDIR/lib/"
```

Comment:
We will publish the jar to maven central. It's fine IMO in this patch to recommend downloading the published jar. And later on we can just suggest linking against the maven coordinates. But I'd like to avoid actually including this JAR in the binary distribution.